viewing_party/party.py

```diff
@@ -1,23 +1,217 @@
+                genre_count[movie["genre"]] += 1
+        most_watched = max(genre_count, key=genre_count.get)
+        return most_watched
+
+
+
+ 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+user_watched= [FANTASY_1,
+                FANTASY_2,
+                FANTASY_3,
+                ACTION_1,
+                INTRIGUE_1,
+                INTRIGUE_2,
+]
+
+first_friend_watched_data = [
+                FANTASY_1,
+                FANTASY_3,
+                FANTASY_4,
+                HORROR_1
+               
+            ]
+second_friend_watched_data = [ FANTASY_1,
+                ACTION_1,
+                INTRIGUE_1,
+                INTRIGUE_3,
+                ]
+def get_unique_watched(user_data):
+    unique_list = [] 
+    comparison_1 =[]
+    comparison_2 =[]
+    for movie in user_data["watched"]:
+        if movie not in first_friend_watched_data:
+            comparison_1.append(movie)
+    for movie in user_data["watched"]:
+        if movie  not in second_friend_watched_data:
+            comparison_2.append(movie)
+    for movie in comparison_2:
+        if movie in comparison_1 and comparison_2:
+           unique_list.append(movie)
+    return unique_list
+    
+
+def get_friends_unique_watched(user_data):
+    friends_unique_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in user_data["watched"]:
+                friends_unique_movies.append(movie)
+    
+    return friends_unique_movies 
+  
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+user_watched = []
+friends_watched = []
+comparison_list =[]
+recommendations = []
+def get_available_recs(user_data):
+    if user_data["watched"] == []:
+        return recommendations
+    else:
+
+        for i in range(len(user_data["watched"])):
+            # user_watched = user_data["watched"][i]
+            user_watched.append(user_data["watched"][i])
+        for i in range(len(user_data["friends"])):
+            for j in range(len(user_data["friends"][i]["watched"])):
+                friends_watched.append(user_data["friends"][i]["watched"][j])
+            # user_watched.append(user_data["watched"][i])
+        for movie in friends_watched:
+            if movie not in user_watched:
+                comparison_list.append(movie)
+        for movie in comparison_list:
+            if movie["host"] in user_data["subscriptions"]:
+                recommendations.append(movie)
+        # print(f"friends watched: {friends_watched}")
+        return recommendations
+    
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+new_list= []
+recommendations_list = []
+def get_new_rec_by_genre(USER_DATA_5):

```

Comment:
we don't want to override `user_data` here. Change the parameter back to `user_data` and think about what helper function you could use here.